tests/model_registry/rest_api/conftest.py

```
@@ -47,3 +71,131 @@ def updated_model_artifact(
+            name=SECURE_MR_NAME,
+            labels=MODEL_REGISTRY_STANDARD_LABELS,
+            grpc={},
+            rest={},
+            istio=ISTIO_CONFIG_DICT,
+            mysql=model_registry_mysql_config,
+            wait_for_resource=True,
+        ) as mr:
+            mr.wait_for_condition(condition="Available", status="True")
+            yield mr
+
+
+@pytest.fixture
+def local_ca_bundle(request: pytest.FixtureRequest, admin_client: DynamicClient) -> Generator[str, Any, Any]:
+    """
+    Creates a local CA bundle file by fetching the CA bundle from a ConfigMap and appending the router CA from a Secret.
+    """
+    namespace = getattr(request, "param", {}).get("namespace", py_config["model_registry_namespace"])
+    ca_bundle_path = getattr(request, "param", {}).get("ca_bundle_path", "ca-bundle.crt")
+    cert_name = getattr(request, "param", {}).get("cert_name", "ca-bundle.crt")
+
+    cm = ConfigMap(client=admin_client, name="odh-trusted-ca-bundle", namespace=namespace, ensure_exists=True)
+    ca_bundle_content = cm.instance.data.get(cert_name)
+    with open(ca_bundle_path, "w", encoding="utf-8") as f:
+        f.write(ca_bundle_content)
+
+    router_secret = Secret(
+        client=admin_client, name="router-ca", namespace="openshift-ingress-operator", ensure_exists=True
+    )
+    router_ca_b64 = router_secret.instance.data.get("tls.crt")
+    if router_ca_b64:
+        router_ca_content = base64.b64decode(router_ca_b64).decode("utf-8")
+        with open(ca_bundle_path, "r", encoding="utf-8") as bundle:
+            bundle_content = bundle.read()
+        if router_ca_content not in bundle_content:
+            with open(ca_bundle_path, "a", encoding="utf-8") as bundle_append:
+                bundle_append.write("\n" + router_ca_content)
+
+    yield ca_bundle_path
+
+    try:
+        os.remove(ca_bundle_path)
+    except FileNotFoundError:
+        pytest.fail(f"CA bundle file not removed: {ca_bundle_path}")
```

This is teardown, we should let it raise exception instead.